overlay.d/05rhcos/usr/lib/dracut/modules.d/41rhcos-check-luks-syntax/rhcos-fail-boot-for-older-luks-syntax

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script tries to look for a `/etc/clevis.json` reference 
+# (used to enable LUKS in RHCOS 4.6 or earlier versions) in an 
+# Ignition config. If it exists and the version of RHCOS is 4.7, 
+# then the script ends with exit 1.
+
+ignition_cfg="/run/ignition.json"
+wanted_path="/etc/clevis.json"
+
+# select the `/etc/clevis.json` entry from a given Ignition config
+if jq -e ".storage.files[] | select(.path==\"${wanted_path}\")" "${ign_config}";  then
+    echo "Failed to boot due to an older syntax used to configure LUKS"
```

Comment:
Let's make sure we give specific information to the user.  Maybe something like "Your Ignition config specifies LUKS filesystem encryption using the obsolete ${wanted_path} config file, which is no longer supported.  Refusing to boot."?

(wrapped to 80 characters, of course)

Reply:
How about `Legacy LUKS configuration is no longer supported. Refusing to boot.`?

Reply:
That could work, but the longer string is clearer, I think.